src/rules/no-unlocalized-strings.ts

```diff
@@ -252,6 +314,24 @@ export const rule = createRule<Option[], string>({
       return node?.name
     }
 
+    function isLiteralInsideJSX(node: TSESTree.Node): boolean {
+      let parent = node.parent
+      let insideJSXExpression = false
+
+      while (parent) {
+        if (parent.type === TSESTree.AST_NODE_TYPES.JSXExpressionContainer) {
+          insideJSXExpression = true
+        }
+        if (parent.type === TSESTree.AST_NODE_TYPES.JSXElement && insideJSXExpression) {
+          return true
+        }
+        parent = parent.parent
+      }
+
+      /* istanbul ignore next */
+      return false
+    }
+
```

Comment:
I suspect it would also match the cases:

```jsx
<div>{myFunction("Hello world")}</div>
```
Because `<div>` is somewhere in ancestor chain for the "Hello world"

I prefer to left `forJsxText` only for `JSXText` nodes. I had an idea to implement autofixes in the future, so this fixes would follow the same logic.

Reply:
It's not... such a test case returns `default` for the error.